pkg/walgen/walgen.go

```diff
@@ -132,7 +124,7 @@ type Set struct {
 }
 
 func (s *Set) Next() bool {
-	if s.curr > len(s.s) {
+	if s.curr >= len(s.s) {
```

Comment:
This was the bug we seen :facepalm:

Reply:
If this code works, it is written by @bwplotka . If not, I don't know 😴